### ISSUE TITLE
Agent version 6.13.0 removes query parameters from original http request uri object when obfuscating 

### ISSUE THREAD

**Issue by VictorJimenezKwast** [newrelic/newrelic-ruby-agent#451]
After upgrading from newrelic_rpm 6.12.0 to 6.13.0 we ran into an issue where our requests seemed to be missing it's query parameters when instrumented by the newrelic agent.  

## Description
In https://github.com/newrelic/newrelic-ruby-agent/pull/404/files#diff-4a15d5cf920433fc7286b4df743a5e6cR17-R25 a change was made to the `filter_uri` method (renamed to obfuscated_uri). The method now calls `parse_and_normalize_url` and the `url.dup`, that was previously present, has been removed. Since the `parse_and_normalize_url` method has some flows in which it does not create a new object it now removes query params (and BA) from the _original_ url object. This breaks the request in many cases.

## Expected Behavior
Obfuscating the request url should not modify the orginal object but always create a new url object instead.

## Steps to Reproduce
Adding a test case to the uri_util_test suite as follows demonstrates the breaking flow:
```
  def test_obfuscate_should_not_modify_input
    original = ::URI.parse("https://foo.com/bar/baz?a=1&b=2#fragment")
    to_obfuscate = original.dup

    NewRelic::Agent::HTTPClients::URIUtil.obfuscated_uri(to_obfuscate).to_s

    assert_equal original, to_obfuscate
  end
```
which results in 
```
  1) Failure:
URIUtilTest#test_obfuscate_should_not_modify_input [test/new_relic/agent/http_clients/uri_util_test.rb:61]:
--- expected
+++ actual
@@ -1 +1 @@
-#<URI::HTTPS https://foo.com/bar/baz?a=1&b=2#fragment>
+#<URI::HTTPS https://foo.com/bar/baz>
```

## Your Environment
newrelic_rpm 6.13.0
ruby 2.6.3
rails 5.2.3

**Comment by VictorJimenezKwast** [comment]
This might also be the root cause of https://github.com/newrelic/newrelic-ruby-agent/issues/450

**Comment by tjallingvanderwal** [comment]
We deployed multiple gem updates when we ran into this problem, and needed to revert the gem updates.

Just updating the NewRelic agent reproduced the problem:

```diff
# Gemfile.lock
-   newrelic_rpm (6.12.0.367)
+   newrelic_rpm (6.13.0)
```

**Comment by mwlang** [comment]
@VictorJimenezKwast Fantastic sleuthing and submission.  I have patched this and expect this to merge to the next release of the agent.  In the meantime, if you'd like to try out the fix, the following in your Gemfile followed by a `bundle install` will use the branch where the patch resides:

```ruby
gem 'newrelic_rpm', 
  git: 'https://github.com/newrelic/newrelic-ruby-agent.git', 
  branch: 'query_params_lost_for_external_requests'
```

The version number will not change, but you should see the branch and commit hash reported in the `bundle install` output: 
```
Using newrelic_rpm 6.13.0 from https://github.com/newrelic/newrelic-ruby-agent.git (at query_params_lost_for_external_requests@43501ed)
```

**Comment by tjallingvanderwal** [comment]
> In the meantime, if you'd like to try out the fix, the following in your Gemfile followed by a bundle install will use the branch where the patch resides:

Verified that the fix solves our problem.
Will keep using 6.12 till the fix is available in a new version though.

**Comment by VictorJimenezKwast** [comment]
@mwlang Thank's for the quick fix.

**Comment by tannalynn** [comment]
Just wanted to let you know that we released the fix for this today in 6.13.1. 
Thank you very much for the repro and test case for this issue.
